kayenta-atlas/src/main/java/com/netflix/kayenta/atlas/model/Backend.java

```diff
@@ -72,12 +78,41 @@
   @Getter
   private String description;
 
+  public List<String> getTargets() {
+    ArrayList<String> ret = new ArrayList<>();
+
+    ret.add(target.replace("$(deployment)", deployment).replace("$(dataset)", dataset));
+
+    if (regions != null) {
+      ArrayList<String> newList = new ArrayList<>();
+      for (String region : regions) {
+        for (String item : ret) {
+          newList.add(item.replace("$(region)", region));
+        }
+      }
+      ret = newList;
+    }
+
+    if (environments != null) {
+      ArrayList<String> newList = new ArrayList<>();
+      for (String environment : environments) {
+        for (String item : ret) {
+          newList.add(item.replace("$(env)", environment));
+        }
+      }
+      ret = newList;
+    }
+
+    return ret.stream().distinct().collect(Collectors.toList());
+  }
+
   public String getUri(String method, String deployment, String dataset, String region, String environment) {
-    String ret = cname
+    String uri = cname
       .replace("$(deployment)", deployment)
       .replace("$(dataset)", dataset)
       .replace("$(region)", region)
       .replace("$(env)", environment);
-    return method + "://" + ret;
+
+    return method + "://" + uri;
```

Comment:
Is this `method` and not `scheme`?

I don't feel strongly about it, but I expected `uri` to contain everything, including the scheme (that is, I don't think it's a uri until the scheme is appended to the front; but I could be wrong).

Maybe s/uri/relativeReference and `return scheme + "://" + relativeReference`.

Or just leave it as is...

Reply:
Will change this.  It is confusing, and I admit I disliked it as I wrote it.